pkg/kubectl/cmd/set/set_image.go

```diff
@@ -226,9 +226,8 @@ func (o *ImageOptions) Run() error {
 			return nil
 		})
 		if transformed && err == nil {
-			// TODO: switch UpdatePodSpecForObject to work on v1.PodSpec, use info.VersionedObject, and avoid conversion completely
 			versionedEncoder := legacyscheme.Codecs.EncoderForVersion(o.Encoder, info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
can remove versionedEncoder and use o.Encoder directly if you're encoding an already versioned object (applies in all places)